development documentation (#578)
scalableminds/webknossos-libs: docs/src/webknossos-py/development.md

  or enter the virtual environment with `poetry shell`.
  The creation of a separate environment can be disabled (e.g. if you want to manage this manually),
  [see here for details](https://python-poetry.org/docs/configuration/#virtualenvscreate).
  To install the preferred version for this repository, run
  [`pip install -f requirements.txt`](https://github.com/scalableminds/webknossos-libs/blob/master/requirements.txt)
* **CI**: We use continuous integration with [github actions](https://github.com/scalableminds/webknossos-libs/actions),
  please see the `CI` workflow for details.
* **Tooling** we try to use across projects (still WIP, see [#581](https://github.com/scalableminds/webknossos-libs/issues/581)):
    * `format.sh`: black and isort
    * `lint.sh`: pylint
    * `typecheck.sh`: mypy
    * `test.sh`: pytest and custom scripts
* **Releases** are triggered via the
  [`Automatic release` github action](https://github.com/scalableminds/webknossos-libs/actions/workflows/release.yml),
  using the `Run workflow` button in the topright corner.
  This updates the changelog and pushes a new tag, which triggers another CI run building and publishing the package.

> ### webknossos package

Reviewer: ```suggestion
### webKnossos package
```

Reviewer 2: Alternatively, maybe use lowercase and backticks when referring to the package name itself

The `webknossos` folder contains examples, which are not part of the package, but are tested via `tests/test_examples.py` and added to the documentation (see `docs/src/webknossos-py/examples`).

> The tests also contain functionality for the webknossos client, sending network requests to a webknossos instance. For normal tests, those requests are read from previous network snapshots using [vcr.py](https://vcrpy.readthedocs.io) via [pytest-recording](https://github.com/kiwicom/pytest-recording).

Reviewer: ```suggestion
The tests also contain functionality for the webKnossos client, sending network requests to a webKnossos instance. For normal tests, those requests are read from previous network snapshots using [vcr.py](https://vcrpy.readthedocs.io) via [pytest-recording](https://github.com/kiwicom/pytest-recording).
```

> This expects a local webknossos setup with specific test data, which is shipped with webknossos. If you're starting and running webknossos manually, please use port 9000 (the default) and run the `tools/postgres/prepareTestDb.sh` script in the webknossos repository (⚠️ this overwrites your local webknossos database). Alternatively, a docker-compose setup is started automatically for the tests, see `test.sh` and `tests/docker-compose.yml` for details.

Reviewer: ```suggestion
This expects a local webKnossos setup with specific test data, which is shipped with webknossos. If you're starting and running webKnossos manually, please use port 9000 (the default) and run the `tools/postgres/prepareTestDb.sh` script in the webKnossos repository (⚠️ this overwrites your local webKnossos database). Alternatively, a docker-compose setup is started automatically for the tests, see `test.sh` and `tests/docker-compose.yml` for details.
```

### wkcuber package

Currently the test setup consists of different scripts as well as pytest tests. The following commands are run in CI:
```bash
tar -xzvf testdata/WT1_wkw.tar.gz
poetry run pytest tests
poetry run tests/scripts/all_tests.sh
```

There's also a `test.sh` which is outdated atm, see [issue #580](https://github.com/scalableminds/webknossos-libs/issues/580).

### cluster_tools package

To test the SLURM setup a docker-compose setup is availble. Please see the the [respective Readme](https://github.com/scalableminds/webknossos-libs/blob/master/cluster_tools/README.md) for details.

## Documentation

> We render a common documentation for the webKnossos Server/Website and webknossos-libs from this repository using [mkdocs](https://www.mkdocs.org/). Source-files for the documentation are stored at `docs/src`:

Reviewer 2: ```suggestion
We render a common documentation for webKnossos itself and webknossos-libs from this repository using [mkdocs](https://www.mkdocs.org/). Source-files for the documentation are stored at `docs/src`:
```

* `docs/src/webknossos`: Server & Website documentation, linked from the [webknossos repository](https://github.com/scalableminds/webknossos) (must be available under `docs/wk-repo`, see below).
* `docs/src/api`: Generated using [pdoc](https://pdoc.dev) from Python docstrings.
* `docs/src/webknossos-py` & `docs/src/wkcuber` Documentation for the respective Python Packages

The structure of the documentation page is given by `docs/mkdocs.yml`.

To generate the documentation locally, use

> ```shell

Reviewer 2: I’d prefer this not as a shell block with commands but rather individual blocks, since they are not to be run sequentially (“use one of”)
